### PR TITLE
Streaming media v2 api

### PR DESCRIPTION
**Pull request by johnjohndoe**
I integrated the 

event-details

### CODE REVIEW

**Comment by ligi** [comment]
Great - looks good!
Unfortunately there are problems with the proguard config - this is what my CI-Server has to say about it:
```
Branch indexing
Connecting to https://api.github.com using /****** (github oauth2)
Cloning the remote Git repository
Cloning repository https://github.com/ligi/CampFahrplan.git
 > git init /var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ@script # timeout=10
Fetching upstream changes from https://github.com/ligi/CampFahrplan.git
 > git --version # timeout=10
using GIT_ASKPASS to set credentials github oauth2
 > git fetch --tags --progress https://github.com/ligi/CampFahrplan.git +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin.url https://github.com/ligi/CampFahrplan.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/ligi/CampFahrplan.git # timeout=10
Fetching upstream changes from https://github.com/ligi/CampFahrplan.git
using GIT_ASKPASS to set credentials github oauth2
 > git fetch --tags --progress https://github.com/ligi/CampFahrplan.git +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin1.url https://github.com/ligi/CampFahrplan.git # timeout=10
Fetching upstream changes from https://github.com/ligi/CampFahrplan.git
using GIT_ASKPASS to set credentials github oauth2
 > git fetch --tags --progress https://github.com/ligi/CampFahrplan.git +refs/pull/*/head:refs/remotes/origin/pr/*
Merging master commit 62b4bc109074a8bc61f17694cf829266ee191d01 into PR head commit 6e3c730f1aac6a578211403886005b95266705f6
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 6e3c730f1aac6a578211403886005b95266705f6
 > git merge 62b4bc109074a8bc61f17694cf829266ee191d01 # timeout=10
 > git rev-parse HEAD^{commit} # timeout=10
Merge succeeded, producing 6e3c730f1aac6a578211403886005b95266705f6
Checking out Revision 6e3c730f1aac6a578211403886005b95266705f6 (PR-7)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 6e3c730f1aac6a578211403886005b95266705f6
First time build. Skipping changelog.
[Pipeline] node
Running on master in /var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ
[Pipeline] {
[Pipeline] stage (checkout)
Using the ‘stage’ step without a block argument is deprecated
Entering stage checkout
Proceeding
[Pipeline] checkout
Cloning the remote Git repository
Cloning repository https://github.com/ligi/CampFahrplan.git
 > git init /var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ # timeout=10
Fetching upstream changes from https://github.com/ligi/CampFahrplan.git
 > git --version # timeout=10
using GIT_ASKPASS to set credentials github oauth2
 > git fetch --tags --progress https://github.com/ligi/CampFahrplan.git +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin.url https://github.com/ligi/CampFahrplan.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/ligi/CampFahrplan.git # timeout=10
Fetching upstream changes from https://github.com/ligi/CampFahrplan.git
using GIT_ASKPASS to set credentials github oauth2
 > git fetch --tags --progress https://github.com/ligi/CampFahrplan.git +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin1.url https://github.com/ligi/CampFahrplan.git # timeout=10
Fetching upstream changes from https://github.com/ligi/CampFahrplan.git
using GIT_ASKPASS to set credentials github oauth2
 > git fetch --tags --progress https://github.com/ligi/CampFahrplan.git +refs/pull/*/head:refs/remotes/origin/pr/*
Merging master commit 62b4bc109074a8bc61f17694cf829266ee191d01 into PR head commit 6e3c730f1aac6a578211403886005b95266705f6
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 6e3c730f1aac6a578211403886005b95266705f6
 > git merge 62b4bc109074a8bc61f17694cf829266ee191d01 # timeout=10
 > git rev-parse HEAD^{commit} # timeout=10
Merge succeeded, producing 6e3c730f1aac6a578211403886005b95266705f6
Checking out Revision 6e3c730f1aac6a578211403886005b95266705f6 (PR-7)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 6e3c730f1aac6a578211403886005b95266705f6
[Pipeline] stage (lint)
Using the ‘stage’ step without a block argument is deprecated
Entering stage lint
Proceeding
[Pipeline] sh
[FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ] Running shell script
+ ./gradlew lintccc33c3Release
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/2.14.1/userguide/gradle_daemon.html.
Download https://jcenter.bintray.com/com/squareup/okhttp3/logging-interceptor/3.5.0/logging-interceptor-3.5.0.pom
Download https://jitpack.io/com/github/johnjohndoe/Brockman/b6cd736ab0a5882b2251b0a62b92f0e69f09827d/Brockman-b6cd736ab0a5882b2251b0a62b92f0e69f09827d.pom
Download https://jcenter.bintray.com/com/squareup/picasso/picasso/2.5.2/picasso-2.5.2.pom
Download https://jcenter.bintray.com/com/squareup/picasso/picasso-parent/2.5.2/picasso-parent-2.5.2.pom
Download https://jcenter.bintray.com/com/squareup/retrofit2/converter-moshi/2.1.0/converter-moshi-2.1.0.pom
Download https://jcenter.bintray.com/com/squareup/moshi/moshi/1.3.1/moshi-1.3.1.pom
Download https://jcenter.bintray.com/com/squareup/moshi/moshi-parent/1.3.1/moshi-parent-1.3.1.pom
Download https://jcenter.bintray.com/com/squareup/okhttp3/logging-interceptor/3.5.0/logging-interceptor-3.5.0.jar
Download https://jcenter.bintray.com/com/squareup/moshi/moshi/1.3.1/moshi-1.3.1.jar
Download https://jcenter.bintray.com/com/squareup/retrofit2/converter-moshi/2.1.0/converter-moshi-2.1.0.jar
Download https://jitpack.io/com/github/johnjohndoe/Brockman/b6cd736ab0a5882b2251b0a62b92f0e69f09827d/Brockman-b6cd736ab0a5882b2251b0a62b92f0e69f09827d.jar
Download https://jcenter.bintray.com/com/squareup/picasso/picasso/2.5.2/picasso-2.5.2.jar
Incremental java compilation is an incubating feature.
:app:preBuild UP-TO-DATE
:app:extractProguardFiles
:app:preCcc33c3ReleaseBuild
:app:checkCcc33c3ReleaseManifest
:app:preCcc33c3DebugBuild UP-TO-DATE
:app:prepareComAndroidSupportAnimatedVectorDrawable2510Library
:app:prepareComAndroidSupportAppcompatV72510Library
:app:prepareComAndroidSupportSupportCompat2510Library
:app:prepareComAndroidSupportSupportCoreUi2510Library
:app:prepareComAndroidSupportSupportCoreUtils2510Library
:app:prepareComAndroidSupportSupportFragment2510Library
:app:prepareComAndroidSupportSupportMediaCompat2510Library
:app:prepareComAndroidSupportSupportV42510Library
:app:prepareComAndroidSupportSupportVectorDrawable2510Library
:app:prepareComGithubLigiExtraCompats04Library
:app:prepareCcc33c3ReleaseDependencies
:app:compileCcc33c3ReleaseAidl
:app:compileCcc33c3ReleaseRenderscript
:app:generateCcc33c3ReleaseBuildConfig
:app:generateCcc33c3ReleaseResValues
:app:generateCcc33c3ReleaseResources
:app:mergeCcc33c3ReleaseResources
:app:processCcc33c3ReleaseManifest
:app:processCcc33c3ReleaseResources
:app:generateCcc33c3ReleaseSources
:app:incrementalCcc33c3ReleaseJavaCompilationSafeguard
:app:compileCcc33c3ReleaseKotlin
:app:compileCcc33c3ReleaseJavaWithJavac
:app:compileCcc33c3ReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:app:copyCcc33c3ReleaseKotlinClasses
:app:compileLint
:app:lintCcc33c3Release
Wrote HTML report to file:///var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/outputs/lint-results-ccc33c3Release.html
Wrote XML report to file:///var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/outputs/lint-results-ccc33c3Release.xml
Lint found 0 errors and 296 warnings

BUILD SUCCESSFUL

Total time: 34.788 secs
[Pipeline] step
[android-lint] Collecting Android Lint files...
[android-lint] Finding all files that match the pattern **/lint-results*.xml
[android-lint] Parsing 1 file in /var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ
[android-lint] Successfully parsed file /var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/outputs/lint-results-ccc33c3Release.xml with 294 unique warnings and 2 duplicates.
[Pipeline] stage (test)
Using the ‘stage’ step without a block argument is deprecated
Entering stage test
Proceeding
[Pipeline] sh
[FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ] Running shell script
+ ./gradlew testccc33c3DebugUnitTest
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/2.14.1/userguide/gradle_daemon.html.
Incremental java compilation is an incubating feature.
:app:preBuild UP-TO-DATE
:app:preCcc33c3DebugBuild UP-TO-DATE
:app:checkCcc33c3DebugManifest
:app:extractProguardFiles
:app:preCcc33c3ReleaseBuild
:app:prepareComAndroidSupportAnimatedVectorDrawable2510Library UP-TO-DATE
:app:prepareComAndroidSupportAppcompatV72510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCompat2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCoreUi2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCoreUtils2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportFragment2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportMediaCompat2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportV42510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportVectorDrawable2510Library UP-TO-DATE
:app:prepareComGithubLigiExtraCompats04Library UP-TO-DATE
:app:prepareCcc33c3DebugDependencies
:app:compileCcc33c3DebugAidl
:app:compileCcc33c3DebugRenderscript
:app:generateCcc33c3DebugBuildConfig
:app:generateCcc33c3DebugResValues
:app:generateCcc33c3DebugResources
:app:mergeCcc33c3DebugResources
:app:processCcc33c3DebugManifest
:app:processCcc33c3DebugResources
:app:generateCcc33c3DebugSources
:app:incrementalCcc33c3DebugJavaCompilationSafeguard
:app:compileCcc33c3DebugKotlin
:app:compileCcc33c3DebugJavaWithJavac
:app:compileCcc33c3DebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:app:copyCcc33c3DebugKotlinClasses
:app:incrementalCcc33c3DebugUnitTestJavaCompilationSafeguard UP-TO-DATE
:app:preCcc33c3DebugUnitTestBuild UP-TO-DATE
:app:prepareCcc33c3DebugUnitTestDependencies
:app:compileCcc33c3DebugUnitTestKotlin
:app:compileCcc33c3DebugUnitTestJavaWithJavac
:app:processCcc33c3DebugJavaRes UP-TO-DATE
:app:processCcc33c3DebugUnitTestJavaRes UP-TO-DATE
:app:compileCcc33c3DebugUnitTestSources
:app:mockableAndroidJar
:app:assembleCcc33c3DebugUnitTest
:app:copyCcc33c3DebugUnitTestKotlinClasses
:app:testCcc33c3DebugUnitTest

BUILD SUCCESSFUL

Total time: 23.609 secs
[Pipeline] step
Recording test results
[Pipeline] publishHTML
[htmlpublisher] Archiving HTML reports...
[htmlpublisher] Archiving at BUILD level /var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/reports/tests to /var/lib/jenkins/jobs/FahrPlan/branches/PR-7/builds/1/htmlreports/UnitTest
[Pipeline] stage (assemble)
Using the ‘stage’ step without a block argument is deprecated
Entering stage assemble
Proceeding
[Pipeline] sh
[FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ] Running shell script
+ ./gradlew assembleccc33c3Release
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/2.14.1/userguide/gradle_daemon.html.
Incremental java compilation is an incubating feature.
:app:preBuild UP-TO-DATE
:app:extractProguardFiles
:app:preCcc33c3ReleaseBuild
:app:checkCcc33c3ReleaseManifest
:app:preCcc33c3DebugBuild UP-TO-DATE
:app:prepareComAndroidSupportAnimatedVectorDrawable2510Library UP-TO-DATE
:app:prepareComAndroidSupportAppcompatV72510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCompat2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCoreUi2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCoreUtils2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportFragment2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportMediaCompat2510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportV42510Library UP-TO-DATE
:app:prepareComAndroidSupportSupportVectorDrawable2510Library UP-TO-DATE
:app:prepareComGithubLigiExtraCompats04Library UP-TO-DATE
:app:prepareCcc33c3ReleaseDependencies
:app:compileCcc33c3ReleaseAidl UP-TO-DATE
:app:compileCcc33c3ReleaseRenderscript UP-TO-DATE
:app:generateCcc33c3ReleaseBuildConfig
:app:generateCcc33c3ReleaseResValues UP-TO-DATE
:app:generateCcc33c3ReleaseResources UP-TO-DATE
:app:mergeCcc33c3ReleaseResources UP-TO-DATE
:app:processCcc33c3ReleaseManifest UP-TO-DATE
:app:processCcc33c3ReleaseResources UP-TO-DATE
:app:generateCcc33c3ReleaseSources
:app:incrementalCcc33c3ReleaseJavaCompilationSafeguard
:app:compileCcc33c3ReleaseKotlin
:app:compileCcc33c3ReleaseJavaWithJavac
:app:compileCcc33c3ReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:app:copyCcc33c3ReleaseKotlinClasses
:app:compileCcc33c3ReleaseNdk UP-TO-DATE
:app:compileCcc33c3ReleaseSources
:app:lintVitalCcc33c3Release
:app:mergeCcc33c3ReleaseShaders
:app:compileCcc33c3ReleaseShaders
:app:generateCcc33c3ReleaseAssets
:app:mergeCcc33c3ReleaseAssets
:app:processCcc33c3ReleaseJavaRes UP-TO-DATE
:app:transformResourcesWithMergeJavaResForCcc33c3Release
:app:transformClassesAndResourcesWithProguardForCcc33c3Release
ProGuard, version 5.2.1
Reading input...
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/com.github.johnjohndoe/Brockman/b6cd736ab0a5882b2251b0a62b92f0e69f09827d/3ceb2a49f3f294a468e00901f9a534d5ad26778d/Brockman-b6cd736ab0a5882b2251b0a62b92f0e69f09827d.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/support-compat/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/support-fragment/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/com.squareup.okio/okio/1.11.0/840897fcd7223a8143f1d9b6f69714e7be34fd50/okio-1.11.0.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/com.squareup.okhttp3/okhttp/3.5.0/4b0bb6bfbecbdd42b97ff400af342bfe7f32bc83/okhttp-3.5.0.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/support-core-utils/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/com.squareup.retrofit2/retrofit/2.1.0/2de7cd8b95b7021b1d597f049bcb422055119f2c/retrofit-2.1.0.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/com.squareup.picasso/picasso/2.5.2/7446d06ec8d4f7ffcc53f1da37c95f200dcb9387/picasso-2.5.2.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/support-core-ui/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/.android-sdk/extras/android/m2repository/com/android/support/support-annotations/25.1.0/support-annotations-25.1.0.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-runtime/1.0.5-3/6c199a70673336eeaf1d5dff1e55a08a63151c36/kotlin-runtime-1.0.5-3.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/support-vector-drawable/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/com.squareup.okhttp3/logging-interceptor/3.5.0/94c11f915bfe1aeaa5ef6351ddfbe58bae1ea397/logging-interceptor-3.5.0.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/support-v4/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/appcompat-v7/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/com.squareup.retrofit2/converter-moshi/2.1.0/9cf9b3171c839a49b1924303105e8b4e57b497e8/converter-moshi-2.1.0.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/animated-vector-drawable/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.github.ligi/ExtraCompats/0.4/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/exploded-aar/com.android.support/support-media-compat/25.1.0/jars/classes.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/com.squareup.moshi/moshi/1.3.1/bd07892e5380af4eba1c8900556cae953815f46d/moshi-1.3.1.jar] (filtered)
Reading program jar [/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.0.5-3/306c2e49fcf7f0f66470565db443bab97612fdea/kotlin-stdlib-1.0.5-3.jar] (filtered)
Reading program jar [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/transforms/mergeJavaRes/ccc33c3/release/jars/2/1f/main.jar] (filtered)
Reading program directory [/var/lib/jenkins/workspace/FahrPlan_PR-7-ZHQI5KUDR7M6LHNYL4757ZDSYFTVYQS4OIIW2RSVOZSDLPK2HZOQ/app/build/intermediates/classes/ccc33c3/release] (filtered)
Reading library jar [/var/lib/jenkins/.android-sdk/platforms/android-25/android.jar]
Reading library jar [/var/lib/jenkins/.android-sdk/platforms/android-25/optional/org.apache.http.legacy.jar]
Note: duplicate definition of library class [org.apache.http.params.HttpConnectionParams]
Note: duplicate definition of library class [org.apache.http.params.HttpParams]
Note: duplicate definition of library class [org.apache.http.params.CoreConnectionPNames]
Note: duplicate definition of library class [org.apache.http.conn.ConnectTimeoutException]
Note: duplicate definition of library class [org.apache.http.conn.scheme.SocketFactory]
Note: duplicate definition of library class [org.apache.http.conn.scheme.HostNameResolver]
Note: duplicate definition of library class [org.apache.http.conn.scheme.LayeredSocketFactory]
Note: duplicate definition of library class [android.net.http.HttpResponseCache]
Note: duplicate definition of library class [android.net.http.SslCertificate]
Note: duplicate definition of library class [android.net.http.SslCertificate$DName]
Note: duplicate definition of library class [android.net.http.SslError]
Note: there were 11 duplicate class definitions.
      (http://proguard.sourceforge.net/manual/troubleshooting.html#duplicateclass)
Initializing...
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Cache
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Cache
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Request$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Request$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Request$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Request$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Request$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Call
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Response
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Response
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.ResponseBody
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Response
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Response
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.ResponseBody
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.ResponseBody
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Cache
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Request$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Response
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Cache
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Request
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Call
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Response
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.ResponseBody
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.OkHttpClient
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.CacheControl
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Request$Builder
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Response
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.ResponseBody
Warning: com.squareup.picasso.OkHttpDownloader: can't find referenced class com.squareup.okhttp.Cache
Warning: retrofit2.Platform$Java8: can't find referenced class java.lang.invoke.MethodHandles$Lookup
Warning: retrofit2.Platform$Java8: can't find referenced class java.lang.invoke.MethodHandles$Lookup
Warning: retrofit2.Platform$Java8: can't find referenced class java.lang.invoke.MethodHandle
Warning: retrofit2.Platform$Java8: can't find referenced class java.lang.invoke.MethodHandle
Warning: retrofit2.Platform$Java8: can't find referenced class java.lang.invoke.MethodHandles
Warning: retrofit2.Platform$Java8: can't find referenced class java.lang.invoke.MethodHandle
Warning: retrofit2.Platform$Java8: can't find referenced class java.lang.invoke.MethodHandles$Lookup
Warning: retrofit2.Platform$Java8: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
Note: com.squareup.moshi.ClassFactory: can't find dynamically referenced class sun.misc.Unsafe
Note: com.squareup.picasso.Utils: can't find dynamically referenced class com.squareup.okhttp.OkHttpClient
Note: kotlin.jvm.internal.Reflection: can't find dynamically referenced class kotlin.reflect.jvm.internal.ReflectionFactoryImpl
Note: okhttp3.internal.platform.AndroidPlatform: can't find dynamically referenced class com.android.org.conscrypt.SSLParametersImpl
Note: okhttp3.internal.platform.AndroidPlatform: can't find dynamically referenced class org.apache.harmony.xnet.provider.jsse.SSLParametersImpl
Note: okhttp3.internal.platform.AndroidPlatform$CloseGuard: can't find dynamically referenced class dalvik.system.CloseGuard
Note: okhttp3.internal.platform.Platform: can't find dynamically referenced class sun.security.ssl.SSLContextImpl
Note: retrofit2.Platform: can't find dynamically referenced class org.robovm.apple.foundation.NSObject
Note: retrofit2.Platform$IOS$MainThreadExecutor: can't find dynamically referenced class org.robovm.apple.foundation.NSOperationQueue
Note: there were 9 unresolved dynamic references to classes or interfaces.
      You should check if you need to specify additional program jars.
      (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclass)
Warning: there were 60 unresolved references to classes or interfaces.
         You may need to add missing library jars or update their versions.
         If your code works fine without the missing classes, you can suppress
         the warnings with '-dontwarn' options.
         (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedclass)
Warning: Exception while processing task java.io.IOException: Please correct the above warnings first.
:app:transformClassesAndResourcesWithProguardForCcc33c3Release FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesAndResourcesWithProguardForCcc33c3Release'.
> Job failed, see logs for details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 23.571 secs
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline

GitHub has been notified of this commit’s build result

ERROR: script returned exit code 1
Finished: FAILURE
Erstelldatum dieser Seite: 28.12.2016 15:28:49 MEZREST APIJenkins ver. 2.35
```

If this is fixed I will be happy to merge and release

**Comment by johnjohndoe** [comment]
@ligi I hope this will resolve the build failure.

File: app/src/main/java/org/ligi/fahrplan/MyApp.java [hunk 1]
```diff
@@ -85,6 +94,15 @@ public void onCreate() {
         lectureList = null;
     }
 
+    public
+    @NonNull
+    OkHttpClient getOkHttpClient() {
```
Comment (ligi) [inline app/src/main/java/org/ligi/fahrplan/MyApp.java]: where is this function used?  In fetchStreamingOffers you create another OKHttp instance

**Comment by ligi** [comment]
:+1:

**Pull request merged**